gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineImmutables.java

```diff
@@ -76,22 +74,12 @@ private static boolean hasImmutablesProcessor(Project project, SourceSet sourceS
         return project
                 .getConfigurations()
                 .getByName(sourceSet.getAnnotationProcessorConfigurationName())
-                .getIncoming()
-                .getResolutionResult()
-                .getAllComponents()
+                .getAllDependencies()
                 .stream()
                 .anyMatch(BaselineImmutables::isImmutablesValue);
     }
 
-    private static boolean isImmutablesValue(ResolvedComponentResult component) {
-        ComponentIdentifier id = component.getId();
-
-        if (!(id instanceof ModuleComponentIdentifier)) {
-            return false;
-        }
-
-        ModuleComponentIdentifier moduleId = (ModuleComponentIdentifier) id;
-
-        return Objects.equals(moduleId.getGroup(), "org.immutables") && Objects.equals(moduleId.getModule(), "value");
+    private static boolean isImmutablesValue(Dependency dependency) {
+        return Objects.equals(dependency.getGroup(), "org.immutables") && Objects.equals(dependency.getName(), "value");
```

Comment:
If people bring in the actual immutables processor jar (not `::annotations` classifier) and actually use it we should still actually apply this? Couldn't we fix this by keeping the code looking at transitives but making sure one of the components has a group `org.immutables`, artifactId `value` _and_ no classifier? Might have to use the `artifacts` rather than `components` api from Gradle to do this from memory?

Reply:
> Couldn't we fix this by keeping the code looking at transitives but making sure one of the components has a group org.immutables, artifactId value and no classifier?

This is what I tried to do at first, but I couldn't figure out how to get at the classifier here. Although it is just more complexity.

If https://github.com/palantir/conjure-java-runtime-api/pull/1099 doesn't fix the issue I'll play around with this a bit more.

Although I have to imagine that bringing in the immutables processor as a transitive dependencies and then using it is pretty unlikely. Honestly, if you're using an annotation processor and not declaring an explicit dependency, then you have no right to complain if your build is slow because we didn't enable incremental compilation for you.

Reply:
I _swear_ I remember seeing the horror of some annotation processor produce an immutables class that was then compiled by immutables... It was a long time ago, sense may have prevailed.

Reply:
> you have no right to complain if your build is slow because we didn't enable incremental compilation for you

I agree with the sentiment but the problem is it just silently becomes slower with no indication why... and devs often just suffer in silence. I guess this is telling me we should probably build something that makes a loud log line or fails the build when it detects and non-incremental annotation processor somehow (in an ideal world).

Reply:
I updated it with the version that checks the classifier using the old APIs (Gradle still does not have a configuration cacheable api for maven classifiers 😢: https://github.com/gradle/gradle/issues/23702).